src/poliastro/core/propagation/__init__.py

```diff
@@ -893,13 +895,51 @@ def danby(k, r0, v0, tof, numiter=20, rtol=1e-8):
                 cta = ecc - np.cosh(E)
 
             nu = np.arctan2(sta, cta)
-            return coe2rv(k, p, ecc, inc, raan, argp, nu)
-
+            break
```

Comment:
The diff got a bit messed up and it's difficult to see, but inside `danby_coe` there's a leftover `return coe2rv` on line 853 (of the current commit):

https://github.com/poliastro/poliastro/pull/1026/files#diff-2fa7459268fa44ca071c0369b2cdde7dR853

Also, notice that removing the `return` statements makes the current logic slightly wrong, because it ends up being something like this:

```python
    if ecc == 0:
        # Solving for circular orbit
        ...
        nu = ...  # We could stop here

    elif ecc < 1.0:
        # For elliptical orbit
        ...
        E = ...  # nu will be computed later

    else:
        # For parabolic and hyperbolic
        ...
        E = ...  # nu will be computed later

    # Iterations begin
    n = 0
    while n <= numiter:
        ...
        if np.abs(f) <= rtol:
            ...
            nu = ...  # FIXME: We compute nu with the `while` loop even if we have it already!
            break

    else:  # Good addition on your side!
        raise ValueError("Maximum number of iterations has been reached.")

    return nu
```

Reply:
Well, I left this long explanation because at first I thought that fixing this logic would be more difficult. But doing `return nu` when it's first available fixes the problem in one line!